compiler/testData/ir/irText/expressions/funInterface/samConversionsWithSmartCasts.fir.kt.txt

```diff
@@ -14,29 +14,29 @@ fun run2(r1: KRunnable, r2: KRunnable) {
 }
 
 fun <T> test0(a: T) where T : KRunnable, T : Function0<Unit> {
-  run1(r = a)
+  run1(r = a /*-> KRunnable */)
```

Comment:
As I can judge FE 1.0 + IR says that here should not be SAM conversion (see `samConversionsWithSmartCasts.kt.txt`)

Reply:
It's actually not the only SAM conversion we don't need. There are many in this test and some other tests as well, and glad this addition(?) caught your attention, thanks!

So, it looks like, such unnecessary conversions have been added for a long time, and this extension to subtype of function type just reveals more cases. Added one more commit to not add SAM conversion for clear subtype cases like this.